first approach to do the Makefile to automatize the setting up

litosnano opened the pull request:
I've tried to find out a good solution to automatize the setting up with a Makefile but right now I'm not capable of doing it. I've made two different approach you'll see in the Makefile (the second of them is #comment it).
A bit of advice it would be more than appreciate it.

Commit:
first approach to do the Makefile to automatize the setting up

adriacidre commented:
Hey @litosnano, Okay, let's first define what our Makefile should do.

## Install
`make install` should basically go through the installation process defined on the readme, this is:
- [ ] go install 
- [ ] create an `~/.aliases` file in case it does not exist
- [ ] append `[ -f ~/.aliases ] && source ~/.aliases` to your `.zshrc` or `. bashrc` file

A Makefile to accomplish this could look something like:
```sh
install:
  go install # this will build the binaries and move the generated binary (named ali) to your GOBIN
  if [[ ! -e ~/.aliases ]]; then # if ~/.aliases does not exist
    touch ~/.aliases # this will create an empty file, we will be back to this later as we want to create this file with some extra content :P
  fi
  echo "[ -f ~/.aliases ] && source ~/.aliases" >> ~/.zshrc # add the necessary line to .zshrc, we will need to also support bashrc etc.. but by now we can go with this option.
```

That's it! The Makefile to install this library should look something like this. I haven't tested it, so you may need to tweak it :P

Lets try to make this simple approach work before going deeper :)

litosnano commented:
Thanks. I'll have a look.

Commit:
some changes in makefile that don't work

litosnano commented:
Sorry but I've tried to find the correct script but it doesn't work. According to shellcheck.net the script is right but it appears two different errors that I don't know how to solve it. I've been looking for information for four days but I've always ended in the same stackoverflow suggestions that it doesn't work as well.

Commit:
create the makefile that introduce at the end of ~/.zshrc the followingcommand: [ -f ~/.aliases ] && source ~/.aliases

litosnano commented:
I've just pushed without the if sentence and it works. Just that every time you call make install the sentence is added at the end of the ~/zshrc file. Hence, if you type three times make install three times the sentence will be added at the end.

adriacidre commented on Makefile:
```diff
@@ -0,0 +1,6 @@
+install:
+	go install
+	touch ~/.aliases
+	echo "[ -f ~/.aliases ] && source ~/.aliases" >> ~/.zshrc
```
You should add a new line at the end of the file.

litosnano replied:
Ok, I've changed the readme and added a new line.
Thanks

adriacidre reviewed (commented):
If `make install` is the way to go to install this library we should reflect this on the README

Commit:
:

Commit:
Readme modified according new setting up of /zshrc by make file

adriacidre commented on README.md:
````diff
@@ -18,18 +18,12 @@ Clone the repository in your computer:
 
 ```
-[ -f ~/.aliases ] && source ~/.aliases
-ali() {
````
The lines below haven't been ported to Makefile, so, you can't remove it.

Ideally the Makefile should create that file *if it doesn't exist* and add the command to `~/.aliases` like:
```sh
# ali
## description: Alias manager
## usage: ali ...
## group: core
ali() {
    $GOPATH/github.com/adriacidre/ali/ali $@	
    source ~/.aliases	
    if [[ "$1" == "rm" ]] then	
        unalias $2	
    fi	
}	
```

adriacidre replied:
On the other hand you can also leave it on the README and go for it on a different pull request.

Commit:
modifying Readme undone a mistake

adriacidre commented on Makefile:
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+install:
+	go install
```
This line will generate and install an executable _(go binary)_ named `ali` on your system, and the user will be able to call id directly with `$ ali`.

The problem is we want `ali` to be the alias described on the README, which internally calls go binary, and as they have the same name it will produce conflicts.

On the other hand this command  _make install_ is to provide the user with a something to work on development. So, instead, we should use `go build` that will be generating the binary on the current folder.

litosnano replied:
Ok, I understand that. I'll change it.

adriacidre commented on README.md:
```diff
@@ -18,18 +18,24 @@ Clone the repository in your computer:
+    $GOPATH/github.com/adriacidre/ali/ali $@	
+    source ~/.aliases	
+    if [[ "$1" == "rm" ]] then	
+        unalias $2	
```
review your IDE as it's adding trailing spaces on each line

litosnano replied:
I'll remove the spaces

adriacidre commented on README.md:
````diff
@@ -18,18 +18,24 @@ Clone the repository in your computer:
+$ make install
+```
+
+And set up the ali with some commands like the ones below with your own $GOPATH
````
This sentence does not make any sense 😹 Let's remove it and automate the alias creation.

litosnano replied:
Ok

adriacidre commented on Makefile:
```diff
@@ -0,0 +1,6 @@
+
+install:
+	go install
+	touch ~/.aliases
```
Create a new folder and file `_tpl/.aliases` with the content
```
# ali
## description: Alias manager
## usage: ali ...
## group: core
ali() {
    {{PATH}}/ali $@	
    source ~/.aliases	
    if [[ "$1" == "rm" ]] then	
        unalias $2	
    fi	
}	
```

Replace the `touch` command by 
```
cp -n _tpl/.aliases ~/.aliases
```
this will create the aliases file with the content we want by default.

Once we have the aliases file in place, we need to replace the `{{PATH}}` string by the path to our binary, in this case the current path, which can be extracted from the command `pwd`. A new line calling sed in the makefile like this should do the trick:
```
...
cp -n _tpl/.aliases ~/.aliases
sed -i 's/{{PATH}}/`pwd`/g' ~/.aliases
... 
```

None of this stuff has been tested, so run all the tests you consider useful

litosnano replied:
I'll do my best.

Commit:
fixing three different issues:
- go build instead of go install due to the conflicts
- delete trailing spaces
- automate the alias creation

Commit:
changing folder name

Commit:
changing the makefile

Commit:
trying to automated the alias creation but fail (install error). Some help here to continue, please.

adriacidre committed:
Makefile to replace path on template, and setup ali
